README.md

```diff
@@ -73,6 +73,30 @@ To install Spectator Sport in your Rails application:
     end
     ```
 
+   If you are using an authentication similar to the one used in ONCE products, you can use an auth constraint in your routes:
+   ```ruby
+   # config/routes.rb
+   class AuthRouteConstraint
+     def matches?(request)
+       return false unless request.session[:user_id]
+       user = User.find request.session[:user_id]
```

Comment:
Totally immaterial to this PR (which is a nice suggestion!), but I believe that the user, once queried, should be put into the `request.env` (`request.env["current_user"] = ...` so that `ApplicationController#current_user` can reuse the results of that query (by pulling it back out of `request.env`) rather than do a 2nd duplicate query. Just like, FYI for your own apps if you follow this pattern 😁

Reply:
Yes, the installation step is probably too long now after this PR. 

Maybe we can use `<details>` / `<summary>` to hide some of the things...

At the current moment, I'm assigning the current_user inside a controller concern and a `helper_method :current_user`

```ruby
#controllers/sessions_controller.rb
def create
  if user = User.active.authenticate_by(email_address: params[:email_address], password: params[:password])
      start_new_session_for user
      redirect_to post_authenticating_url
   else
      render_rejection :unauthorized
   end
end


#controllers/concerns/auth/session_auth.rb
def current_user
  Current.user
end

private

def start_new_session_for(user)
  user.sessions.start!(user_agent: request.user_agent, ip_address: request.remote_ip).tap do |session|
    authenticated_as session
  end
end
    
def authenticated_as(session)
  Current.user = session.user
  request.session[:user_id] = session.user.id
  set_authenticated_by(:session)
  cookies.signed.permanent[:session_token] = { value: session.token, httponly: true, same_size: :lax }
end
```

Thank you for the suggestion. 
I should refactor this and set the **request.env["current_user"]** to avoid duplicate queries for the same thing. This was the fastest thing I found at that time to get going.

Reply:
> Maybe we can use `<details> / <summary>` to hide some of the things...

Just my opinion, but I am totally ok with a super long readme. I think maintaining the ability to `cmd/ctrl-f` to search is better than the comparable tradeoff of hiding/minimizing content selectively.

I think it's own section and simply an `#anchor-link` is good enough.

> Thank you for the suggestion.

No prob. Kind of random 😁 It was something I sweated over myself when I moved off Devise (totally forgot [I wrote this up](https://island94.org/2024/01/replacing-devise-with-rails-has_secure_password-and-friends))

Reply:
That was a great read with lots of insights. Thank you!
I created a new PR and moved the authorization code into a separate section.

I forgot to thank you very much for approving the PR!